src/conversions.rs

```diff
@@ -508,7 +509,10 @@ impl<C: Clone, T: FromJSValConvertible<Config=C>> FromJSValConvertible for Vec<T
 
             for i in 0..length {
                 let mut val = RootedValue::new(cx, UndefinedValue());
-                assert!(JS_GetElement(cx, obj.handle(), i, val.handle_mut()));
+                if !JS_GetElement(cx, obj.handle(), i, val.handle_mut()) {
+                    // On JS exception or error exit loop
```

Comment:
I thought this couldn't fail, but I double-checked, and it can:
```js
Object.defineProperty(Array.prototype, 1, {get: function() {throw 7}});
new Blob([,,,])
```